toolchain/check/testdata/interop/cpp/typedef.carbon

```diff
@@ -0,0 +1,138 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// INCLUDE-FILE: toolchain/testing/testdata/min_prelude/int.carbon
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/interop/cpp/typedef.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/interop/cpp/typedef.carbon
+
+// --- typedef.h
+
+typedef int foo;
+
+using bar = int;
+
+// --- use_typedef.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp library "typedef.h";
+
+fn F() {
+  //@dump-sem-ir-begin
+  var n: Cpp.foo = 42;
+  var m: Cpp.bar = 76;
+  //@dump-sem-ir-end
```

Comment:
Can we also do:

```suggestion
  var m: Cpp.bar = 76;
  // Types are equivalent.
  var l: Cpp.bar* = &n;
```

And maybe something for class typedefs (since `i32` is a special-case), such as:

```
// C++:

class C {};
typedef C D;

// Carbon:

fn F(m: Cpp.C, n: Cpp.D) {
  // Verifies type equivalence.
  var o: Cpp.C* = &m;
  var p: Cpp.D* = &n;
}
```

Reply:
Yes, that's what I intended. Thanks. :)